sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -714,15 +718,47 @@ public WritableColumnVector arrayData() {
   public WritableColumnVector getChild(int ordinal) { return childColumns[ordinal]; }
 
   /**
-   * Returns the elements appended.
+   * Returns the number of child vectors.
+   */
+  public int getNumChildren() {
+    return childColumns.length;
+  }
+
+  /**
+   * Returns the elements appended. This is useful
    */
   public final int getElementsAppended() { return elementsAppended; }
 
+  /**
+   * Increment number of elements appended by 'num'.
+   *
+   * This is useful when one wants to use the 'putXXX' API to add new elements to the vector, but
+   * still want to keep count of how many elements have been added (since the 'putXXX' APIs don't
+   * increment count).
+   */
+  public final void addElementsAppended(int num) {
+    elementsAppended += num;
+  }
+
   /**
    * Marks this column as being constant.
    */
   public final void setIsConstant() { isConstant = true; }
 
+  /**
+   * Marks this column only contains null values.
+   */
+  public final void setAllNull() {
```

Comment:
1. `column.numNulls()` may still 0 after we call  `setAllNull()`
2.  do `column.setAllNull()` and `column.putXXX`, then `column.isAllNull` still true.

The above two scenes look strange

Reply:
This is similar to `setIsConstant` though: you can also call `column.putXXX` after the call.

Reply:
OK, let's ignore this for the time being